### ISSUE TITLE
Callback URL - Migration from single server to HA cluster.

### ISSUE THREAD

**Issue by Rucknar** [rancher/rancher#4426]
**Summary:**
Callback URL not consistent, can cause confusion and errors.
**Rancher Version:**
1.0.1
**Docker Version:**
1.10.3
**OS:** 
RancherOS 0.4.4
**Steps to Reproduce:** 
* Create new Rancher server instance on Host1, running on port 8080 with a remote SQL server.
* Create ELB to route traffic to this and the other hosts we will use shortly. At this stage, the callback URL is `http:///myelb:8080/`
* Initiate HA configuration, URL now becomes `https://myelb/` in this setup window.
* Run the scripts etc, Rancher becomes HA (yay!)
* Remove the port 8080 binding from the ELB listeners (no longer needed now https is setup).
* Rancher-compose-executer and go-machine-service start spinning, you get errors in the logs they are not able to connect to `http://myelb:8080/`
* Re-add the port to the ELB and the services recover.

**Results:** 
Some services seem to be bound to the original callback URL
**Expected:**
All services use the

**Comment by Rucknar** [comment]
The above was actually causing the addition of AWS hosts to fail, further investigation showed that some of the containers in the 'System HA' stack were recyling almost constantly. The two mentioned earlier specifically, i deleted and purged these and they came up using the right URL this time.

I'm 100% the callback URL was set to the default during the initial install, then the new one with https was specified in the HA setup. Somehow these containers started with the old values. Regardless, the workaround is to delete/purge and start new versions of the containers.

**Comment by deniseschannon** [comment]
For the initial host that you used to create the scripts, you had stopped and removed Rancher server and ran the new scripts on it, correct?

Just want to make sure that step was completed.

**Comment by Rucknar** [comment]
Yes it was, however i'll admit i'm not sure how far along in the sequence it was done. It wasn't clear from the instructions if i had to do that server first, or the others first then that server...

**Comment by LLParse** [comment]
Hi @Rucknar, I believe the issue is caused because the callback URL wasn't updated to the HTTPS:// LB endpoint before generating the HA script. I don't know if we even mention that in the docs, but I agree that it is confusing and error-prone.

Ideally our HA setup will be push-button in the future, but for the time being we will update the docs. Thanks for catching this.

**Comment by Rucknar** [comment]
@LLParse Not sure you could switch it to HTTPS at that stage, at least i thought i checked that before running the HA scripts.

**Comment by LLParse** [comment]
Well, that piece of configuration isn't relevant until HA is started and you are registering hosts to the UI cluster. I agree that communication can't occur with the HTTPS endpoint until the System HA stack stabilizes, which is a bit confusing since you're required to configure it beforehand.

As long as you trust in the system HA stack to eventually serve traffic on port 443, you're good to go.

**Comment by Rucknar** [comment]
Re-testing this on a fresh build again. This time, i've run the rancher-ha.sh script on the primary node first and waited for it to complete before starting it on other hosts. 

All seems to be better this time, i've still got a 'add-host' issue on AWS, i'll cover that in the other ticket. 

**Minor Bug:** initiate HA configuration, this goes to a page showing 'waiting for agents' or similar, but unless you refresh the command `sudo bash ./rancher-ha.sh rancher/server:v1.0.1` isn't displayed at the top. 

Ideally, the instructions provided through the upgrade process could be a little clearer here, suspect i won't be the only one who hits issues with the process.
